nestor_api/lib/k8s/builders.py

```diff
@@ -46,6 +47,19 @@ def get_anti_affinity_zone(app_config: dict, process_name: str, templates: dict)
     return anti_affinity_zone
 
 
+def get_image_name(app_config, process_name, options):
+    """Returns the definitive image name"""
+    image_tag = options["tag"]
+    branch = options.get("branch")
+
+    if branch is not None:
+        image_tag = branch
```

Comment:
What about ?
```suggestion
    image_tag = options["branch"] if "branch" in options else options["tag"]
```
Or you could also change the function's parameter to receive `ref` directly